msteams-platform/m365-apps/teamsjs-support-m365.md

```diff
@@ -0,0 +1,2069 @@
+            <td></td>
+        </tr>
+    </tbody>
+</table>
+
+## `mail`
+
+[Reference](/javascript/api/@microsoft/teams-js/mail)
+
+Namespace providing email-related functionality.
+
+<br />
+<table border>
+    <thead>
+        <tr>
+            <th></th>
+            <th colspan=3>Web</th>
+            <th colspan=3>Desktop</th>
+            <th colspan=6>Mobile</th>
+        </tr>
+        <tr>
+            <th></th>
+            <th colspan=3></th>
+            <th colspan=3>Windows</th>
+            <th colspan=3>Android</th>
+            <th colspan=3>iOS</th>
+        </tr>
+        <tr>
+            <td></td>
+            <td><img alt="Teams" src="./images/teams-icon.png"/></td>
+            <td><img alt="Microsoft 365 app" src="./images/microsoft-365-icon.png"/></td>
+            <td><img alt="Outlook" src="./images/outlook-icon.png"/></td>
+            <td><img alt="Teams" src="./images/teams-icon.png"/></td>
+            <td><img alt="Microsoft 365 app" src="./images/microsoft-365-icon.png"/></td>
+            <td><img alt="Outlook" src="./images/outlook-icon.png"/></td>
+            <td><img alt="Teams" src="./images/teams-icon.png"/></td>
+            <td><img alt="Microsoft 365 app" src="./images/microsoft-365-icon.png"/></td>
+            <td><img alt="Outlook" src="./images/outlook-icon.png"/></td>
+            <td><img alt="Teams" src="./images/teams-icon.png"/></td>
+            <td><img alt="Microsoft 365 app" src="./images/microsoft-365-icon.png"/></td>
+            <td><img alt="Outlook" src="./images/outlook-icon.png"/></td>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <th>mail</th>
+            <td></td>
+            <td></td>
+            <td>&#x2713;</td>
+            <td></td>
+            <td></td>
+            <td>&#x2713;</td>
+            <td></td>
+            <td></td>
+            <td></td>
+            <td></td>
+            <td></td>
+            <td></td>
+        </tr>
+    </tbody>
+</table>
+
+## `media`
+
+[Reference](/javascript/api/@microsoft/teams-js/media)
+
+Namespace providing image file-related functionality.
+
+App capabilities that require the user to grant [device permissions](../concepts/device-capabilities/device-capabilities-overview.md) (such as *media*) aren't yet supported for apps running outside of Teams. There is currently no way to check app permissions in Settings or your app header when running in Outlook or Microsoft 365 app. If a Teams app running in Microsoft 365 app or Outlook calls a TeamsJS (or HTML5) API that triggers device permissions, that API will generates an error and fail to display a system dialog asking for user consent.
+
+Current guidance is to modify your code to catch the failure:
+
+* Check [isSupported()](../tabs/how-to/using-teams-client-library.md#differentiate-your-app-experience) on a capability before using it. Some capabilities, such as `media` and `meeting` don't yet support *isSupported* calls and don't yet work outside of Teams.
```

Comment:
>Check [isSupported()](../tabs/how-to/using-teams-client-library.md#differentiate-your-app-experience) on a capability before using it. Some capabilities, such as `media` [](http://example.com/codeflow?start=2&length=169)

It seems kind of weird to offer this as guidance in the section of the media capability since it literally says that it doesn't work with the media capability.

Reply:
Eww...yes, let me check with Alicia about more actionable verbiage for this section.